pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -108,7 +109,9 @@ func New(address string, port uint, runtime string, rootPath string) (Interface,
 
 	if _, err := os.Stat(rootPath); err != nil {
 		if os.IsNotExist(err) {
-			return nil, fmt.Errorf("rootDirectory %q does not exist", rootPath)
+			if err := os.MkdirAll(path.Clean(rootPath), 0750); err != nil {
```

Comment:
@dims Refer to comment in [#50709](https://github.com/kubernetes/kubernetes/issues/50709#issuecomment-322646497).